### DISCUSSION TITLE
Suggestions on how to create a "meta package"

### DISCUSSION THREAD

**Discussion by beriberikix** [Homebrew/discussions#1219] [Tap maintenance and Homebrew development]
Hello! I'm working on a project that has multiple packages and would like them to both be individually installable but be installed in the same "package root" directory. For example, we have `FOO-BAR` & `FOO-BAZ`. The user may `brew install FOO-BAR` and `brew install FOO-BAZ` and they both be in a `/foo` directory. This is generally so we can set `FOO_INSTALL_DIR` to `/foo` and another application can reference it.

Any suggestions on how we might go about doing that? Any examples in the wild with similar requirements?

**Comment by carlocab** [comment]
Installing separate formulae into the same prefix isn't really compatible with `brew`'s install sandbox. The simplest way to do this might be to write a single formula `foo` that has the options `with-bar` and `with-baz`.

**Reply by beriberikix** [reply]
Thanks for the quick response! So with optional steps the user would execute `brew install foo --bar` as opposed to `brew install foo-bar`? Two challenges with that approach is that it's a bit unnatural flow and as we increase the number of packages (currently 9,) it'll start to get unwieldy.

That's certainly an option! But would like to explore others. Any thing else that comes to mind?

**Reply by SMillerDev** [reply]
That's really your only option since homebrew doesn't have the concept of meta-packages. The only other option would be to make a pull request to introduce the concept.

**Reply by carlocab** [reply]
Another option might be to create three formulae:
```
foo-bar-baz
foo-bar
foo-baz
```
and perhaps a fourth one that's just `foo`, if that's meaningful. But that might even be more unwieldy than a single formula with many options.

**Reply by beriberikix** [reply]
That was my original thought too, but they wouldn't be in the same root folder, right?

Another idea would be to use symlinks. For example, as a final stage of installation we create a symlink to something like `~/.foo`. That way we can 1) maintain the sandbox model 2) allow users to `brew install foo-bar` normally & 3) we can point our other tools to `FOO_INSTALL_DIR`.

WDYT? I also need to see if our community is okay creating a dotfolder for our tools.

**Reply by carlocab** [reply]
They wouldn't, but I think the idea is that you would only have one of `foo-bar-baz`, `foo-bar`, or `foo-baz` installed at any one time, so your meta-package will always live in the same directory.

You won't be able to write those symlinks at install time because of the sandbox. It restricts writing anywhere that isn't the formula's prefix or the (temporary) build directory.

I guess you could create a "meta-formula" that installs only symlinks to where the sub-formulae would be installed. Otherwise, you can ask users to create those symlinks after the install (maybe this can be automated with a script on first run).
